### PR TITLE
fix macos drop privs

### LINKED ISSUE #205: Not dropping root privileges running on macOS 11.6.1
Good afternoon,

I've just been testing `boringtun` on macOS but noticed that when I run using the following command, that it does not drop root privileges.

sudo WG_QUICK_USERSPACE_IMPLEMENTATION=boringtun WG_SUDO=1 wg-quick down ~/wireguard/example.conf

The `boringtun` process is running as root.

Cheers,
-Chris

### PR DESCRIPTION
**Pull request by phillyqueso**
fixes #205 

references:
- https://blog.lxsang.me/post/id/28 <-- nudged to use [nix](https://docs.rs/nix/0.18.0/nix/unistd/index.html).
- https://github.com/capistrano/capistrano/issues/1520 <-- helped understand the underlying macos specific issue.

### CODE REVIEW

**Comment by Noah-Kennedy** [comment]
@phillyqueso run `cargo fmt`

**Comment by Noah-Kennedy** [comment]
@phillyqueso so I have good news and bad news. The good news is that we fixed the CI issues on the master branch, and all you need to do here is merge in those changes. The bad news is that there are some mean merge conflicts here, as we just split the crate into a workspace, which should have been done long ago.

**Comment by phillyqueso** [comment]
> @phillyqueso so I have good news and bad news. The good news is that we fixed the CI issues on the master branch, and all you need to do here is merge in those changes. The bad news is that there are some mean merge conflicts here, as we just split the crate into a workspace, which should have been done long ago.

No problem @Noah-Kennedy! I'll rebase and push the fix soon.

**Review by Noah-Kennedy — APPROVED** [review]
Thanks for the PR!

File: boringtun/src/device/drop_privileges.rs [hunk 1]
```diff
@@ -5,23 +5,35 @@ use crate::device::errno_str;
...
+use nix::unistd::User;
+
 pub fn get_saved_ids() -> Result<(uid_t, gid_t), Error> {
     // Get the user name of the sudoer
-    let uname = unsafe { getlogin() };
-    if uname.is_null() {
-        return Err(Error::DropPrivileges("NULL from getlogin".to_owned()));
+    #[cfg(target_os = "macos")]
+    {
+        let uname: &'static str = env!("USER");
```
Comment (jiegec) [inline boringtun/src/device/drop_privileges.rs:15]: env!() retrieves the environment variable at compile time, is it expected? If I compile the code in one user, and use the binary in another user(or machine), the behavior is incorrect.

Comment (Noah-Kennedy) [reply boringtun/src/device/drop_privileges.rs:15]: I'll open up a PR after I ear breakfast.

Comment (Noah-Kennedy) [reply boringtun/src/device/drop_privileges.rs:15]: Also, next time, it's much easier to track this if you create an issue to track the problem.

Comment (Noah-Kennedy) [reply boringtun/src/device/drop_privileges.rs:15]: Fixed

**Pull request merged**